shared/entity/type.go

```diff
@@ -2,19 +2,24 @@ package entity
 
 import (
 	"fmt"
-	"net/url"
-	"runtime"
-	"strings"
-
-	"github.com/canonical/lxd/shared"
-	"github.com/canonical/lxd/shared/api"
-	"github.com/canonical/lxd/shared/logger"
-	"github.com/canonical/lxd/shared/version"
 )
 
 // Type represents a resource type in LXD that is addressable via the API.
 type Type string
 
+// typeImpl represents common attributes an entity type must have.
```

Comment:
As this is not exported please can we use the full name, im not sure what "impl" means here, is it "implementation"?

Reply:
See https://go.dev/talks/2014/names.slide#13 also for hints on interface naming

> When an interface includes multiple methods, choose a name that accurately describes its purpose (examples: net.Conn, http.ResponseWriter, io.ReadWriter).

Reply:
Yeah I was expecting feedback on this :laughing: I wasn't sure what to call it. Do you have any suggestions? My thoughts were:
- `typeImpl`, short for `typeImplementation`
- `typeImplementation`
- `typeProperties`

Reply:
Yeah typeProperties, typeBehaviours, or typeInfo I reckon

Reply:
`typeInfo` sounds more familiar in the LXD code base so I'll go this that.